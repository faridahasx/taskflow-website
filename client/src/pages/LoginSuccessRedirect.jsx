import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import LoadingPage from "../components/Loading/LoadingPage";

const LoginSuccessRedirect = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  useEffect(() => {
    localStorage.setItem("firstLogin", "true");
    dispatch({ type: "IS_LOGGED", payload: true });
    // dispatch({ type: "ALERT", payload: "Login Success!" });
    navigate("/");
    // eslint-disable-next-line
  }, []);

  return <LoadingPage />;
};

export default LoginSuccessRedirect;