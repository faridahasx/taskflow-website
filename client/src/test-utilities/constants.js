import { MISSING_INPUT_FIELD } from "constants/alertMessages";

// AUTH
export const passwordLabel = "Password:";
export const emailLabel = "Email:";
export const firstnameLabel = "First Name:";
export const lastnameLabel = "Last Name:";

// Server responses
export const successResponse = { status: 200, data: "Success" };

// Redux state
export const reduxStateOnMissingInputField = [
  {
    type: "ALERT",
    payload: MISSING_INPUT_FIELD,
  },
];
