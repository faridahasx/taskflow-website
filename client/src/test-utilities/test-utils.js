import { render } from "@testing-library/react";
import { TasksDispatchContext } from "context/TaskContext";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

const customRender = (ui, { props, ...options } = {}) => {
  const { store, mockDispatchTasks } = props;

  const AllTheProviders = ({ children }) => (
    <Provider store={store}>
      <TasksDispatchContext.Provider value={mockDispatchTasks}>
        <BrowserRouter>{children}</BrowserRouter>
      </TasksDispatchContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
// export * from "@testing-library/jest-dom";
export { customRender as render };
