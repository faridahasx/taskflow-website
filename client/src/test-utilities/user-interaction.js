import { fireEvent, screen } from "./test-utils";

const clickByText = (text) => fireEvent.click(screen.getByText(text));
const clickByTitle = (title) => fireEvent.click(screen.getByTitle(title));
const clickByTestId = (title) => fireEvent.click(screen.getByTestId(title));

const changeByLabelText = (label, value) => {
  const input = screen.getByLabelText(label);
  fireEvent.change(input, { target: { value } });
  return input;
};

const changeByTestId = (id, value) => {
  const input = screen.getByTestId(id);
  fireEvent.change(input, { target: { value } });
  return input;
};

const pressKeyboardButton = (container, buttonName) => {
  fireEvent.keyDown(container, { key: buttonName });
};

const pressEnter = (container) => {
  pressKeyboardButton(container, "Enter");
};

const pressEscape = (container) => {
  pressKeyboardButton(container, "Escape");
};

export {
  clickByText,
  clickByTitle,
  clickByTestId,
  changeByTestId,
  changeByLabelText,
  pressEnter,
  pressEscape,
};
