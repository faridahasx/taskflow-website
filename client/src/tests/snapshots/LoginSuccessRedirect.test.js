import { render } from "test-utilities/test-utils";
import { useNavigate } from "react-router-dom";
import mockInitialState, {
  mockStore,
} from "test-utilities/mocks/mockReduxState";
import LoginSuccessRedirect from "pages/LoginSuccessRedirect";

test("LoginSuccessRedirect snapshot", () => {
  useNavigate.mockReturnValue(jest.fn());

  const { asFragment } = render(<LoginSuccessRedirect />, {
    props: { store: mockStore(mockInitialState) },
  });
  expect(asFragment()).toMatchSnapshot("LoginSuccessRedirect");
});
