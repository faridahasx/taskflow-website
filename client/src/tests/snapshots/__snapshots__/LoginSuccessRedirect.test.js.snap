// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSuccessRedirect snapshot: LoginSuccessRedirect 1`] = `
<DocumentFragment>
  <div>
    <div
      class="page-loading-indicator"
    >
      <span
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
        />
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
